src/main/java/com/google/cloud/tools/maven/DeployMojo.java

```diff
@@ -93,9 +93,11 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     // execute stage
     super.execute();
 
-    if (deployables.size() == 0) {
-      deployables.add(
-          new File(getStagingDirectory() + "/app.yaml"));
+    File appYamlFile = new File(getStagingDirectory() + "/app.yaml");
+    if (deployables.size() == 0 && appYamlFile.exists()) {
+      deployables.add(appYamlFile);
+    } else {
+      deployables.add(stagingDirectory);
```

Comment:
I don't think we want to maintain yet another app.yaml generation template.
Here we're just letting the Cloud SDK generate the app.yaml based on its fingerprinting logic.